internal/parse/parser.go

```diff
@@ -43,3 +45,66 @@ func (p *Parser) Parse(input string) (*ParsedExpr, error) {
 type ParsedExpr struct {
 	queryParts []QueryPart
 }
+
+// peekByte returns true if the current byte
+// equals the one passed as parameter.
+func (p *Parser) peekByte(b byte) bool {
+	return p.pos < len(p.input) && p.input[p.pos] == b
+}
+
+// skipByte jumps over the current byte if it matches
+// the byte passed as a parameter. Returns true in that case, false otherwise.
+func (p *Parser) skipByte(b byte) bool {
+	if p.pos < len(p.input) && p.input[p.pos] == b {
+		p.pos++
+		return true
+	}
+	return false
+}
+
+// skipByteFind advances the parser until it finds a byte that matches the one
+// passed as parameter and then jumps over it. In that case returns true. If the
+// end of the string is reached and no matching byte was found, it returns
+// false.
+func (p *Parser) skipByteFind(b byte) bool {
+	for i := p.pos; i < len(p.input); i++ {
+		if p.input[i] == b {
+			p.pos = i + 1
+			return true
+		}
+	}
+	return false
+}
+
+// skipSpaces advances the parser jumping over consecutive spaces. It stops when
+// finding a non-space character. Returns true if the parser position was
+// actually changed, false otherwise.
+func (p *Parser) skipSpaces() bool {
+	mark := p.pos
+	for p.pos < len(p.input) {
+		if p.input[p.pos] != ' ' {
+			break
+		}
+		p.pos++
+	}
+	return p.pos != mark
+}
+
+// skipString advances the parser and jumps over the string passed as parameter.
+// In that case returns true, false otherwise.
+// This function is case insensitive.
+func (p *Parser) skipString(s string) bool {
+	if p.pos+len(s) <= len(p.input) &&
+		strings.EqualFold(p.input[p.pos:p.pos+len(s)], s) {
+		p.pos += len(s)
+		return true
+	}
+	return false
+}
+
+// isNameByte returns true if the byte passed as parameter is considered to be
```

Comment:
Good point.
`isNameByte` is used (in the experiment) from `parseIdentifier` which is used to recognize things like `foo.bar` in things that come before the `AS` keyword *and* things that come after it.
Apparently SQLite [deviates a bit](https://www.sqlite.org/draft/tokenreq.html) from the standard too. Sybase IQ supports things like the pipe (`|`) in column names, postgres has its own rules, etc.
It's going to be difficult to be DB-agnostic here. Maybe we can allow for everything to be a name byte except for _blank separators_ (spaces, tabs, newlines...).

Reply:
That sounds reasonable, it'd probably be best to specify specifically what certainly isn't a name rather than what is. As long long as the parser doesn't rely on the function too heavily

Reply:
> it'd probably be best to specify specifically what certainly isn't a name

That creates double negatives which are harder to work with ("is this not-not-part-of-a-name?").

I think the main issue here is the lack of more context: a name of what?   I suspect we don't want to use the full SQL naming convention, but rather something that will be friendlier to our environment. But again, it depends on the "name of what?" idea.

Reply:
Here I think of `name` as something we can use to represent:
* table name
* column name
* Go type name
* Go struct field name

So we can use it to know when to stop parsing and say, "okay, this is where the name ends". So the function is really simple in the sense that it only accepts alphanumerics and underscore. This will be a limitation mainly for table/column names since [I don't expect](https://go.dev/ref/spec#identifier) any Go code to have weird characters in the type names or struct field names. But column names can go wild in some other databases. I'm ok limiting this to some sensible defaults.

Reply:
Go names don't use underscores in general, and it may also be problematic to *stop parsing* based on that specifically, because as @Aflynn50 mentioned above, the syntax is richer than what we consider "good naming". To be clear, the constraint on good names itself is a good idea, but it's worth considering how to apply it. For the time being, I suggest dropping it and then bringing it back together with the PR that uses it, so we have a bit more context for the conversation.

Reply:
Removed.